### PR TITLE
Add sorting and filtering to flags page

### LINKED ISSUE #4400: Add sorting and filtering to flags page
In Prometheus, this PR https://github.com/prometheus/prometheus/pull/8988 has been merged.
It provides a search field and a way to sort the different flag.

image

The PR also changed the `fuzzy` lib in favor of a new up to date one `@nexucis/fuzzy`.

Since it has been merged on Prometheus, it can make sense to push these changes as well in Thanos to still be synchronized

### PR DESCRIPTION
**Pull request by Nexucis**
fix #4400

### CODE REVIEW

**Comment by Nexucis** [comment]
If I'm not wrong, the review is more on your side @onprem right ? :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 
 ### Added
 
-- [#4384](https://github.com/thanos-io/thanos/pull/4384) Fix the experimental PromQL editor when used on multiple line.
+- [#4403](https://github.com/thanos-io/thanos/pull/4403) Add sorting and filtering to flags page
```
Comment (onprem) [inline CHANGELOG.md:15]: Small nit 
```suggestion
- [#4403](https://github.com/thanos-io/thanos/pull/4403) UI: Add sorting and filtering to flags page
```

**Review by onprem — APPROVED** [review]
Just a small nit in the CHANGELOG, the code looks perfect. Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (Nexucis) [reply CHANGELOG.md]: fixed :). thanks for the review !

**Comment by yeya24** [comment]
@Nexucis Can you please rebase? Thanks for the contribution!

**Comment by Nexucis** [comment]
@yeya24 done :)

**Comment by Nexucis** [comment]
@yeya24 you need to reapprove the PR if you enable the auto merge :)

**Review by yeya24 — APPROVED** [review]
LGTM

**Pull request merged**